### ISSUE TITLE
Review remarks

### ISSUE THREAD

**Issue by Northengard** [a1k0u/os#1]
Great job, but there is some little remarks
In Readme.md
14) You say that "позволит переменной сохраниться до перезагрузки системы." It's not quite correct: these variables will be lost even if user restarts his terminal.
15) /home/scripts - it seems your username is missed or this subdirectory has been created in shared user space, which is kinda harmful if you do not plan to share its content with other users. The same refers to all subdirs created not inside your user home directory, but in /home space
21, 24) please, left a reference on the datasets, you've been using
31) Well done, but this may not work if username is long: for example
![169394517-2b9dcc0c-519a-4b87-9b3b-6b6bd8595126](https://user-images.githubusercontent.com/43110253/169405558-6c98dd70-c8d0-45d9-9fae-c4656786089f.png)
Command should be corrected.
28) does not work for me in case of recursive search. + where should I add -r flag? From my point of view this task can be solved without pipe
49) according to the task description script should check if there is any file as input argument to output
As if you share your apy key may check API environment variable in runtime and set it in case of nonexistence
`os.environ["API_OPENWEATHER"]="27950c473536d6bb06394814058f4575"`
Unfortunately this script fails for me on openweather request but it's not your fault, they just has some `issues` which is beyond our work.
51) https://github.com/a1k0u/os/blob/64c3c3137e4cd5524bedd671d867ef0dd01cafb8/book.py#L51
kwargs means Key Word ARGumentS, commonly it goes with with ** which means the list of arguments passed by name, which consists of variables that may be optional, so:
1) kwargs mentioning in docstring is redundant. 
2) you should pass kwargs with ** notation in function declaration.
3) kwargs should be the last argument in function definition, to draw attention to desired custom arguments that should be passed anyway
FYI: https://habr.com/ru/company/ruvds/blog/482464/
Some words about database stuff: In this case it does not matter but it's a good practice to shield the requests data gathered from user, to prevent on db atacs. https://habr.com/ru/post/148151/ I think it may be usefull to know

**Comment by a1k0u** [comment]
14. I've transformed it into "переменной сохраниться до перезагрузки терминала".
15. Thx for advice, I made pattern for it: "export PATH=$PATH:/home/<some_root>/scripts".

21, 24) link for dataset was added

31)  Wow, it was unsuspected thing. I've fixed it for more fixable situation. `top -n 1 -b | grep -E "${USER:0:7}[+]?"`

**Comment by a1k0u** [comment]
28) Yep... Unfortunately, I made a few mistakes in this assignment. I should use the `-R` flag with `ls` to recursively output folders to `/lib`. `ls -R /lib | grep -E -o "*\.so"`

49, 51) I spent a lot of time learning how to write professional Python code (hint type, Python standard library). To be true, I have to refactor my code to make the codebase architecture more readable and flexible, handle exceptions, remove unnecessary document strings and variable hints.

I've been looking at good Python coding practices here: https://www.youtube.com/watch?v=dKxiHlZvULQ
It was really useful because I learned how to use `typing`, `enum`, `dataclasses` correctly, and also understood the generics, stub files.

Thanks for the information about SQL injections, I already knew about it, but I guess I was too lazy to process some incorrect data at that time. 

So, next time I'm going to write code more professionally, run tests and write awesome documentation.

Thank you for your question!
